### PR TITLE
Fix #89 Fix #91 Improve ModelServerClient for easier customization

### LINKED ISSUE #89: Improve ModelServerClient member visibilites for easier customization
Currently many helper methods are private, their visibility should be changed to improve customization of the `ModelServerClient`.

### LINKED ISSUE #91: Request server/configure does not return JsonResponse anymore
The DefaultServerController should send a success JsonResponse if workspace initialization is completed.
Returning no response at all will cause the following error:
```
ERROR Request configure failed with error: invalid json response body at http://localhost:8081/api/v1/server/configure reason: Unexpected end of JSON input FetchError: invalid json response body at http://localhost:8081/api/v1/server/configure reason: Unexpected end of JSON input
```

### PR DESCRIPTION
**Pull request by ndoschek**
Signed-off-by: Nina Doschek <ndoschek@eclipsesource.com>

### CODE REVIEW

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java [hunk 1]
```diff
@@ -59,17 +59,17 @@
 
 public class ModelServerClient implements ModelServerClientApi<EObject>, ModelServerPaths, AutoCloseable {
 
-   private static final Set<String> DEFAULT_SUPPORTED_FORMATS = ImmutableSet.of(ModelServerPathParameters.FORMAT_JSON,
+   protected static final Set<String> DEFAULT_SUPPORTED_FORMATS = ImmutableSet.of(ModelServerPathParameters.FORMAT_JSON,
```
Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:62]: I think there is no need to make constants protected. Non-private constants should be public for better accesibility.

Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:62]:
```suggestion
   public static final Set<String> DEFAULT_SUPPORTED_FORMATS = ImmutableSet.of(ModelServerPathParameters.FORMAT_JSON,
```

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java [hunk 2]
```diff
@@ -59,17 +59,17 @@
...
       ModelServerPathParameters.FORMAT_XMI);
-   private static final String PATCH = "PATCH";
-   private static final String POST = "POST";
+   protected static final String PATCH = "PATCH";
```
Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:64]:
```suggestion
   public static final String PATCH = "PATCH";
```

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java [hunk 3]
```diff
@@ -59,17 +59,17 @@
...
+   protected static final String POST = "POST";
```
Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:65]:
```suggestion
   public static final String POST = "POST";
```

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java [hunk 4]
```diff
@@ -59,17 +59,17 @@
...
 
-   private static Logger LOG = Logger.getLogger(ModelServerClient.class.getSimpleName());
+   protected static Logger LOG = Logger.getLogger(ModelServerClient.class.getSimpleName());
```
Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:67]: Shouldn't subclasses declare their own logger instance?

File: bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultServerController.java [hunk 5]
```diff
@@ -41,6 +41,7 @@ public void configure(final Context ctx) {
       try {
          if (updateServerConfiguration(newConfiguration)) {
             modelRepository.initialize();
+            // ContextResponse.success(ctx);
```
Comment (tortmayr) [inline bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultServerController.java:44]: Seems like an uncessary line to me

**Review by tortmayr — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java (hunk 4 above)
Comment (ndoschek) [reply bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:67]: Yes, you're right, that makes more sense! I changed it back to private

File: bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultServerController.java (hunk 5 above)
Comment (ndoschek) [reply bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultServerController.java:44]: Oh my bad, this was the whole point of the fix, I added this line again in the next commit :+1:

File: bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java (hunk 1 above)
Comment (ndoschek) [reply bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/ModelServerClient.java:62]: Thanks, yes that makes more sense!

**Review by tortmayr — APPROVED** [review]
LGTM. Thanks for the quick update :+1:

**Pull request merged**